src/java.base/share/classes/java/lang/ProcessBuilder.java

```diff
@@ -1120,6 +1139,22 @@ private Process start(Redirect[] redirects) throws IOException {
                 event.pid = process.pid();
                 event.commit();
             }
+            // Racy initialization for logging; errors in configuration may throw exceptions
+            System.Logger logger = LOGGER;
+            if (logger == null) {
+                LOGGER = logger = System.getLogger("java.lang.ProcessBuilder");
+            }
+            if (logger.isLoggable(System.Logger.Level.DEBUG)) {
+                boolean detail = logger.isLoggable(System.Logger.Level.TRACE);
+                var level = (detail) ? System.Logger.Level.TRACE : System.Logger.Level.DEBUG;
+                var cmdargs = (detail) ? String.join(" ", cmdarray) : cmdarray[0];
+                RuntimeException stackTraceEx = new RuntimeException("ProcessBuilder.start() debug");
+                LOGGER.log(level, "ProcessBuilder.start(): " +
+                                "cmd: " + cmdargs +
+                                ", dir: " + dir +
+                                ", pid: " + process.pid(),
```

Comment:
It may make sense to revert the printout order, print pid first, cwd second, then the args, since argument vectors can get very large. Does Logger limit the line length?

Furthermore - but up to you - when I do printouts like this I always enclose the command args in quotes. That helps people analyze situations like accidental trailing spaces in arguments.

Pity that we cannot show at least PATH and LD_LIBRARY_PATH/LIBPATH.